File: Accounting/src/main/java/com/experiment/accounting/config/ResourceServerConfig.java
package com.experiment.accounting.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter;
import org.springframework.security.oauth2.config.annotation.web.configurers.ResourceServerSecurityConfigurer;
import org.springframework.security.oauth2.provider.OAuth2Authentication;
import org.springframework.security.oauth2.provider.token.RemoteTokenServices;
import org.springframework.security.oauth2.provider.token.ResourceServerTokenServices;

import java.util.Map;

@Configuration
@EnableResourceServer
public class ResourceServerConfig extends ResourceServerConfigurerAdapter {

    @Override
    public void configure(ResourceServerSecurityConfigurer resources) {
        resources.resourceId("api");
    }

    @Bean
    public ResourceServerTokenServices createResourceServerTokenServices() {
        RemoteTokenServices tokenServices = new RemoteTokenServices();
        tokenServices.setCheckTokenEndpointUrl("http://localhost:8090/oauth/check_token");
        tokenServices.setClientId("testClientId");
        tokenServices.setClientSecret("secret");
        return tokenServices;
    }



}

File: Inventory/src/main/kotlin/com/experiment/inventory/config/ResourceServerConfig.kt
package com.experiment.inventory.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer
import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter
import org.springframework.security.oauth2.config.annotation.web.configurers.ResourceServerSecurityConfigurer
import org.springframework.security.oauth2.provider.token.RemoteTokenServices

@Configuration
@EnableResourceServer
class ResourceServerConfig: ResourceServerConfigurerAdapter() {

    override fun configure(resources : ResourceServerSecurityConfigurer){
        resources.resourceId("api")
    }

    @Bean
    fun createResourceServerTokenServices():RemoteTokenServices{
        val tokenServices = RemoteTokenServices()
        tokenServices.setCheckTokenEndpointUrl("http://localhost:8090/oauth/check_token")
        tokenServices.setClientId("testClientId")
        tokenServices.setClientSecret("secret")
        return tokenServices
    }
}